exchanges/oex/oex_wrapper.go

```diff
@@ -0,0 +1,390 @@
+package oex
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Start starts the Oex go routine
+func (o *Oex) Start(wg *sync.WaitGroup) {
+	wg.Add(1)
+	go func() {
+		o.Run()
+		wg.Done()
+	}()
+}
+
+// Run implements the Oex wrapper
+func (o *Oex) Run() {
+	if o.Verbose {
+		log.Debugf("%s polling delay: %ds.\n", o.GetName(), o.RESTPollingDelay)
+		log.Debugf("%s %d currencies enabled: %s.\n", o.GetName(), len(o.EnabledPairs), o.EnabledPairs)
+	}
+	exchangeCurrencies, err := o.GetAllPairs()
+	if err != nil {
+		log.Errorf("%s Failed to get available symbols.\n", o.GetName())
+	}
+	var newExchangeCurrencies currency.Pairs
+	var tempPair currency.Pair
+	for x := 0; x < len(exchangeCurrencies.Data); x++ {
+		tempPair = currency.NewPairFromString(exchangeCurrencies.Data[x].Symbol)
+		newExchangeCurrencies = append(newExchangeCurrencies, exchange.FormatCurrency(tempPair))
+	}
+	err = o.UpdateCurrencies(newExchangeCurrencies, false, true)
```

Comment:
This should be o.UpdateCurrencies(newExchangeCurrencies, false, false)

The first param being the new currencies, the second for whether the currencies are being updated for the enabled or available currencies (in this case for available currencies) and third for whether a forced update is required (in this case it isn't required)